huaweicloud/services/waf/resource_huaweicloud_waf_cloud_instance.go

```diff
@@ -0,0 +1,481 @@
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the virtual gateway is located.",
+			},
+			"resource_spec_code": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(SpecCodeIntroduction),
+					string(SpecCodeStandard),
+					string(SpecCodeProfessional),
+					string(SpecCodePlatinum),
+				}, false),
+				Description: "The specification of the cloud WAF.",
+			},
+			"charging_mode": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"prePaid",
+				}, false),
+				RequiredWith: []string{
+					"period_unit", "auto_renew",
+				},
+				Description: "The charging mode of the cloud WAF.",
+			},
+			"period_unit": common.SchemaPeriodUnit(nil),
+			"period":      common.SchemaPeriod(nil),
+			"auto_renew":  common.SchemaAutoRenewUpdatable(nil),
+			"bandwidth_expack_product": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Elem:        expackProductSchema(),
+				MaxItems:    1,
+				Description: "The configuration of the bandwidth extended packages.",
+			},
+			"domain_expack_product": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Elem:        expackProductSchema(),
+				MaxItems:    1,
+				Description: "The configuration of the domain extended packages.",
+			},
+			"rule_expack_product": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				Elem:        expackProductSchema(),
+				MaxItems:    1,
+				Description: "The configuration of the rule extended packages.",
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The ID of the enterprise project to which the cloud WAF belongs.",
+			},
+			// Attributes
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The current status of the cloud WAF.",
+			},
+		},
+	}
+}
+
+func buildExtendedPackages(packages []interface{}) *clouds.ExpackProductInfo {
+	if len(packages) < 1 {
+		return nil
+	}
+
+	expack := packages[0].(map[string]interface{})
+	return &clouds.ExpackProductInfo{
+		ResourceSize: expack["resource_size"].(int),
+	}
+}
+
+func getAutoRenewValue(autoRenew string) bool {
+	if autoRenew == "" {
+		return false
+	}
+	result, _ := strconv.ParseBool(autoRenew)
+	return result
+}
+
+func resourceCloudInstanceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	wafClient, err := cfg.WafV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating WAF v1 client: %s", err)
+	}
+
+	opts := clouds.CreateOpts{
+		ProjectId:   cfg.HwClient.ProjectID,
```

Comment:
```suggestion
		ProjectId:   wafClient.ProviderClient.ProjectID,
```

you should use the project ID in the service client

Reply:
done.